test/ComponentInstallerTest.php

```diff
@@ -571,4 +571,65 @@ public function testOnPostPackageUninstallCanRemovePackageArraysFromConfiguratio
         $this->assertNotContains('Some\Component', $config);
         $this->assertNotContains('Other\Component', $config);
     }
+
+    public function testModuleIsAppended()
+    {
+        $this->createApplicationConfig(
+            '<' . "?php\nreturn [\n    'modules' => [\n        'Some\Component',\n    ]\n];"
+        );
+
+        $package = $this->prophesize(PackageInterface::class);
+        $package->getName()->willReturn('some/module');
+        $package->getExtra()->willReturn(['zf' => [
+            'module' => 'Some\\Module',
+        ]]);
+
+        $operation = $this->prophesize(InstallOperation::class);
+        $operation->getPackage()->willReturn($package->reveal());
+
+        $event = $this->prophesize(PackageEvent::class);
+        $event->isDevMode()->willReturn(true);
+        $event->getOperation()->willReturn($operation->reveal());
+
+        $this->io->ask(Argument::that(function ($argument) {
+            if (! is_array($argument)) {
+                return false;
+            }
+
+            if (! strstr($argument[0], "Please select which config file you wish to inject 'Some\Module' into")) {
+                return false;
+            }
+
+            if (! strstr($argument[1], 'Do not inject')) {
+                return false;
+            }
+
+            if (! strstr($argument[2], 'application.config.php')) {
+                return false;
+            }
+
+            return true;
+        }), 0)->willReturn(1);
```

Comment:
For anybody else coming across this, here's the explanation for why `1` is correct.

The `$ask` array has three items in it. The first is the question "Please select which config file you wish to inject", and has no option assigned to it. The next two are `Do not inject`, which has index `0`, and `application.config.php`, which has index `1`. The code above this is ensuring we get all three entries, and then returns the value `1`, indicating the user selected `application.config.php`.

Basically, ignore the indices within the method argument verification; what matters are the indices when creating the arguments within the `ComponentInstaller` class itself.